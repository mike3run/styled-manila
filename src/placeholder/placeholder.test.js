import React from 'react'
import styled from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { placeholder } from '..'

test('placeholder works!', () => {
  const Test = styled.div`
    ${placeholder`
      color: pink;
    `}
  `
  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()

  expect(tree).toHaveStyleRule('color', 'pink', {
    modifier: ':placeholder-shown'
  })

  expect(tree).toHaveStyleRule('color', 'pink', {
    modifier: '::-webkit-input-placeholder'
  })
})
