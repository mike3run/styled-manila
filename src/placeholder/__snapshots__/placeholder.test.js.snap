// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder works! 1`] = `
.c0::-webkit-input-placeholder {
  color: pink;
}

.c0::-moz-placeholder {
  color: pink;
}

.c0:-ms-input-placeholder {
  color: pink;
}

.c0::-ms-input-placeholder {
  color: pink;
}

.c0:placeholder-shown {
  color: pink;
}

<div
  className="c0"
/>
`;
